list.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// This file is used to show the list of container checkpoints
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sync"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+func list(cmd *cobra.Command, args []string) error {
+	allPaths := append([]string{defaultCheckpointPath}, additionalCheckpointPaths...)
+
+	var wg sync.WaitGroup
+
+	table := tablewriter.NewWriter(os.Stdout)
+	header := []string{
+		"Namespace",
+		"Pod",
+		"Container",
+		"Time Created",
+		"Checkpoint Name",
+	}
+
+	table.SetHeader(header)
+	table.SetAutoMergeCells(false)
+	table.SetRowLine(true)
+
+	for _, checkpointPath := range allPaths {
+		fmt.Printf("Listing checkpoints in path: %s\n", checkpointPath)
+
+		files, err := filepath.Glob(filepath.Join(checkpointPath, "checkpoint-*"))
+		if err != nil {
+			return err
+		}
+
+		if len(files) == 0 {
+			fmt.Println("No checkpoints found in this path.")
+			continue
+		}
+
+		for _, file := range files {
+			wg.Add(1)
+
+			go func(file string) {
+				defer wg.Done()
```

Comment:
Could we simplify this to something like the following? This also formats the timestamp to make it easier to read.
```Go
for _, file := range files {
	namespace, pod, container, timestamp, err := parseCheckpointName(filepath.Base(file))
	if err != nil {
		log.Printf("Error parsing checkpoint name %s: %v\n", file, err)
		continue
	}

	timeValue, err := time.Parse(time.RFC3339, timestamp)
	if err != nil {
		log.Printf("Error parsing timestamp %s for file %s: %v\n", timestamp, file, err)
		continue
	}

	row := []string{
		namespace,
		pod,
		container,
		timeValue.Format(time.RFC822),
		filepath.Base(file),
	}

	table.Append(row)
}
```